pymodbus/server/async_io.py

```diff
@@ -593,11 +579,8 @@ def stop(cls):
         if not cls.active_server.loop.is_running():
             Log.info("ServerStop called with loop stopped.")
             return
-        asyncio.run_coroutine_threadsafe(cls.async_stop(), cls.active_server.loop)
-        if os.name == "nt":
-            time.sleep(10)
-        else:
-            time.sleep(0.1)
+        future = asyncio.run_coroutine_threadsafe(cls.async_stop(), cls.active_server.loop)
+        future.result(timeout=10 if os.name == 'nt' else 0.1)
```

Comment:
this one I like, there are a number of places in the code, where this can be used.

Reply:
https://github.com/pymodbus-dev/pymodbus/pull/2034

Most other places in the code have either been removed already, or are only `0.1` seconds.